app/models/user.rb

```diff
@@ -1968,7 +1968,9 @@ def set_default_categories_preferences
       end
     end
 
-    CategoryUser.insert_all!(values) if values.present?
+    if values.present?
+      CategoryUser.insert_all(values, unique_by: :idx_category_users_user_id_category_id)
```

Comment:
Referencing an index of another table in the `User` model is quite strange because I think the details about the indexes of a table should be considered an internal concern. Are we able to do an `UPSERT ... ON CONCLICT DO NOTHING` here instead?

Reply:
Or we can do `unique_by: %i[user_id category_id]`. Isn't `ON CONFLICT DO NOTHING` will ignore all the exceptions?

Reply:
Yup that sounds much better :+1: